### PR TITLE
#11 Add support to include and/or exclude list of countries

### LINKED ISSUE #11: Add support for filtering result by list of countries and only return location if the location is inside that list of country

### PR DESCRIPTION
**Pull request by chirag4semandex**
- Added small test for the same

Closes #11

### CODE REVIEW

**Review by djmakwana — APPROVED** [review]

File: mordecai3/elastic_utilities.py [hunk 1]
```diff
@@ -273,16 +273,31 @@ def add_es_data(ex, conn, max_results=50, fuzzy=0, limit_types=False,
                              "fields": ['name', 'asciiname', 'alternativenames'],
                              "type": "phrase"}}
 
+    include_country_filter = None
+    if include_countries is not None and include_countries:
```
Comment (ianrandman) [inline mordecai3/elastic_utilities.py:277]: Why both conditions?

Comment (chirag4semandex) [reply mordecai3/elastic_utilities.py:277]: Checking for `None` & `non empty list`

Comment (ianrandman) [reply mordecai3/elastic_utilities.py:277]: You can keep just the latter condition to handle both for this and `exclude_countries`.

File: mordecai3/elastic_utilities.py [hunk 2]
```diff
@@ -214,7 +214,7 @@ def _clean_search_name(search_name):
     return search_name
 
 def add_es_data(ex, conn, max_results=50, fuzzy=0, limit_types=False,
-                remove_correct=False, known_country=None):
+                remove_correct=False, include_countries:list[str] | None=None, exclude_countries:list[str] | None=None):
```
Comment (ianrandman) [inline mordecai3/elastic_utilities.py:217]: Add spaces surrounding `=` of default values. Add type hints for other parameters.

I also generally suggest to move each parameter to its own line when you have a lot of them.

File: mordecai3/elastic_utilities.py [hunk 3]
```diff
@@ -322,12 +336,14 @@ def add_es_data(ex, conn, max_results=50, fuzzy=0, limit_types=False,
 
 
 def add_es_data_doc(doc_ex, conn, max_results=50, fuzzy=0, limit_types=False,
-                    remove_correct=False, known_country=None):
+                    remove_correct=False, include_countries:list[str] | None=None,
+                    exclude_countries:list[str] | None=None):
```
Comment (ianrandman) [inline mordecai3/elastic_utilities.py:340]: same here

File: mordecai3/geoparse.py [hunk 4]
```diff
@@ -325,12 +325,13 @@ def pick_event_loc(self, d):
...
-                     debug=False, 
-                     trim=True, 
-                     known_country=None,
+    def geoparse_doc(self,
+                     text,
+                     plover_cat=None,
+                     debug=False,
+                     trim=True,
+                     include_countries:list[str] | None=None,
+                     exclude_countries: list[str] | None = None,
```
Comment (ianrandman) [inline mordecai3/geoparse.py:334]: and here

If there is a type hint, there should be spaces surrounding `=` of default value. If no type hint, then no spaces.

File: mordecai3/elastic_utilities.py (hunk 1 above)
Comment (chirag4semandex) [reply mordecai3/elastic_utilities.py]: Sure, done

File: mordecai3/elastic_utilities.py (hunk 2 above)
Comment (chirag4semandex) [reply mordecai3/elastic_utilities.py]: Sure, done

File: mordecai3/elastic_utilities.py (hunk 3 above)
Comment (chirag4semandex) [reply mordecai3/elastic_utilities.py]: Sure, done

File: mordecai3/geoparse.py (hunk 4 above)
Comment (chirag4semandex) [reply mordecai3/geoparse.py]: Done

**Review by ianrandman — APPROVED** [review]

**Pull request merged**